spec/components/auth/instagram_authenticator_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+describe Auth::InstagramAuthenticator do
+
+  it "takes over account if email is supplied" do
+    auth = Auth::InstagramAuthenticator.new
+
+    user = Fabricate(:user)
+
+    auth_token = {
+      info: { email: user.email },
+      uid: "123",
+      provider: "instagram"
+    }
+
+    result = auth.after_authenticate(auth_token)
+
+    expect(result.user.id).to eq(user.id)
+
+    info = UserAssociatedAccount.find_by(provider_name: "instagram", user_id: user.id)
+    expect(info.info["email"]).to eq(user.email)
+  end
+
+  it 'can connect to a different existing user account' do
+    authenticator = Auth::InstagramAuthenticator.new
+    user1 = Fabricate(:user)
+    user2 = Fabricate(:user)
```

Comment:
Same as above, I think we might use want to `fabricate` for `user1` once at the top of this spec.